### DISCUSSION TITLE
aixm dependency - incoherencies

### DISCUSSION THREAD

**Discussion by kcris** [wmo-im/iwxxm#278] [Q&A] [answered]
Hi,

In general, iwxxm depends on aixm

-------------

I noticed that the new airmet v 3.1 - https://github.com/wmo-im/iwxxm/blob/master/IWXXM/airmet.xsd
uses  `schemaLocation="http://www.aixm.aero/schema/5.1.1/AIXM_Features.xsd"`
Looking at the aixm xsd, I see 
* version 5.1.1 
* copyright 2016, 
* a big AIXM_Features.xsd

-------------

The old airmet v 3.0 - https://github.com/wmo-im/iwxxm/blob/v3.0.0/IWXXM/airmet.xsd
uses `schemaLocation="http://www.aixm.aero/schema/5.1.1_profiles/AIXM_WX/5.1.1b/AIXM_Features.xsd"`
Looking at the aixm xsd I see
* version 5.1.1 = same as above
* copyright 2019 (more recent, even if the iwxxm 3.0 is actually older) - not important though
* a small AIXM_Features.xsd

-------------

So I am totally confused. I see twice aixm 5.1.1, but totally different size/contents (and the wrong dates)
So I am not sure if this is a problem or not for iwxxm users.
What am I missing here? these aixm files are referenced by iwxxm schemas.

Thanks

**Comment by blchoy** [comment, marked as answer]
There used to be a discussion on this topic (#199).  To put the long story short, `http://www.aixm.aero/schema/5.1.1_profiles/AIXM_WX/5.1.1b/AIXM_Features.xsd` is a profile not only containing just some of the features in `http://www.aixm.aero/schema/5.1.1/AIXM_Features.xsd` but a number of these features have also been trimmed down (less elements/attributes/enumeration entries, etc.).

The profile was created after the original schema, that's why it carries a later creation date.

Theoretically this new profile should be given a new namespace instead of `http://www.aixm.aero/schema/5.1.1` but unfortunately due to historical reasons this cannot be done.  That's why we switched back to the full schema to import necessary AIXM features to buy time and pave way for a better solution.

**Comment by kcris** [comment]
thanks a lot, that explains everything.

* The trimmed down profile (desirable to minimize iwxxm's dependencies) *without* a new namespace is a bad idea in the xml universe. 
* Fixed now on the xsd side (switched back to true, full aixm 5.1.1 schemas)

I think using both iwxxm 3,0 and iwxxm 3.1 will still require extra care in the client code since element types that are actually-different (old aixm/wx profile vs full aixm now) are still using the same target namespace (`http://www.aixm.aero/schema/5.1.1`) and same name.

Thank you

**Comment by kcris** [comment]
To complement my previous message:

To get a minimal xsd for iwxxm, two things are required
* new namespace
* new elements (minimal set), recursively, ideally considering *all* dependencies not just aixm (gml, gsr, gco, isotc)
Basically it means: start-from-scratch, forget about reusing existing opengis+aixm schemas. Right?

MIght never fully happen, but you were somewhat on the right track with iwxxm 3.0 trying to *minimize* aixm usage.

Not using a new namespace was bad, but the minimization idea was good. 
(since only a small part of those imported xsd are actually being used in iwxxm, right?)

@blchoy  So why not take the opportunity and *just fix the namespace* issue in iwxxm 3.1?
* make one more! copy of old aixm wx profile (which contains a minimal set of aixm types needed by iwxxm)
* and just fix the namespace (a new namespace for the aixm contents) in this new copy of the old aixm-wx-profile

which is cleanly doable in the *new* iwxxm version. Does not break any compatibility since this is a new iwxxm.
And still allows to (properly this time) use the minimal aixm subset.
Why not?

Thank you
